python/lsst/summit/utils/consdbClient.py

```diff
@@ -0,0 +1,412 @@
+        **kwargs: `dict`
+            Additional key/value pairs, overriding ``values``.
+
+        Returns
+        -------
+        response: `requests.Response`
+            HTTP response from the server, with 200 status for success.
+        """
+        if values:
+            values.update(kwargs)
+        else:
+            values = kwargs
+        data = {"values": values}
+        url = _urljoin(
+            self.url,
+            "flex",
+            quote(instrument),
+            quote(obs_type),
+            "obs",
+            quote(str(obs_id)),
+        )
+        if allow_update:
+            url += "?u=1"
+        return self._handle_post(url, data)
+
+    def insert(
+        self,
+        instrument: str,
+        table: str,
+        obs_id: int,
+        values: dict[str, Any],
+        *,
+        allow_update=False,
+        **kwargs,
+    ) -> requests.Response:
+        """Insert values into a single ConsDB fixed metadata table.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        table: `str`
+            Name of the table to insert into.
+        obs_id: `int`
+            Unique observation id.
+        values: `dict` [ `str`, `Any` ], optional
+            Dictionary of column/value pairs to add for the observation.
+        allow_update: `bool`, optional
+            If ``True``, allow replacement of values of existing columns.
+        **kwargs: `dict`
+            Additional column/value pairs, overriding ``values``.
+
+        Returns
+        -------
+        response: `requests.Response`
+            HTTP response from the server, with 200 status for success.
+        """
+        if values:
+            values.update(kwargs)
+        else:
+            values = kwargs
+        data = {"table": table, "obs_id": obs_id, "values": values}
+        url = _urljoin(
+            self.url,
+            "insert",
+            quote(instrument),
+            quote(table),
+            "obs",
+            quote(str(obs_id)),
+        )
+        if allow_update:
+            url += "?u=1"
+        return self._handle_post(url, data)
+
+    def query(self, query: str) -> Table:
+        """Query the ConsDB database.
+
+        Parameters
+        ----------
+        query: `str`
+            A SQL query (currently) to the database.
+
+        Returns
+        -------
+        result: `Table`
+            An ``astropy.Table`` containing the query results.
+
+        Notes
+        -----
+        This is a very general query interface because it is expected that
+        a wide variety of types of queries will be needed. If some types prove
+        to be common, syntactic sugar could be added to make the simpler.
+        """
+        url = _urljoin(self.url, "query")
+        data = {"query": query}
+        result = self._handle_post(url, data).json()
+        return Table(rows=result["data"], names=result["columns"])
```

Comment:
I see that the obs_ids are getting cast to `str` on insertion. Does this bring them back to being ints on reading? If not, I think that would be worth adding, to make sure the type is preserved while round-tripping (especially on the primary key).

Reply:
They're cast to `str` to put in the URL, but they're handled as `int` otherwise, including here.